api/.gitignore

```diff
@@ -1,11 +0,0 @@
-# Dependencies
-node_modules/
-
-# Build artifacts
-build/
-
-# Logs
-npm-debug.log
-
-# API env
-.env
```

Comment:
Okay, this was a combination of ignorance and preference.  
I would normally have a single .gitignore at the root level, and put all the rules there, so you would have
api/.env
but not 
ui/.env
To me it seems simpler to manage them all in one place.
However, in line with your desire to make the modules as independent as possible, I can set it back to how it was and correct the other change which was made previously.

Reply:
What do you think about renaming the ui/.env file to something else, like "env" ?  
My thinking on this is that I _really_ don't want to accidentally check in a file with sensitive information because then you need to muck with git to permanently remove it.

If ui/.env is not sensitive information and it's okay to check it in, then it should be okay to make it a regular file, too.  The dotenv package can do this:
require('dotenv').config({ path: '/full/custom/path/to/your/env/vars' })

Reply:
> What do you think about renaming the ui/.env file to something else, like "env" ?
My thinking on this is that I really don't want to accidentally check in a file with sensitive information because then you need to muck with git to permanently remove it.

I agree that this feels weird if you're used to using env variables on the backend, but variables defined in .env on the frontend will be included in the build of the code and available to the public, so should contain no secrets. The backend .env can contain secrets, but the frontend .env should not. You can see the recommendation about checking .env on the frontend into source control in the create-react-app documentation in the UI README, and read more about it [here](https://github.com/facebook/create-react-app/issues/2560).